tools/mksurfdata_esmf/README

```diff
@@ -0,0 +1,100 @@
+================
+> cd <user/work/area>
+> git clone https://github.com/ESCOMP/CTSM.git <dir_name>
+> cd <dir_name>
+> git checkout feature/new_mksurfdata
+
+# Save time by checking out ccs_config only
+> ./manage_externals/checkout_externals ccs_config
+
+=======================
+building the executable
+=======================
+> cd tools/mksurfdata_esmf
+> rm -rf bld  # (or the contents of /bld) before rerunning CC=...
+> mkdir ./bld
+> cd bld
+> module load intel  # I found I need the latest version of intel
+> module load python  # in case need latest version of python
+> module load cmake
+> bash
+> CC=mpicc FC=mpif90 cmake -DCMAKE_BUILD_TYPE=debug ../src/
+> make VERBOSE=1
+> export MKSURFDATA_BLD=`pwd`  # needed once; execute in /bld
+
+# IF YOU MODIFY .F90 files, start here first, though more than once
+# I have found it necessary to go back to the "rm -rf bld" step
+> cd ../src
+> make clean  # REMEMBER to be in bash
+> make VERBOSE=1
+
+=======================
+running for a single submission: 
+=======================
+> cd ../
+# to generate your target namelist:
+> ./gen_mksurfdata_namelist.py --help
+# for example try --res 1.9x2.5 --start-year 1850 --end-year 1850:
+> ./gen_mksurfdata_namelist.py --res <resolution> --start-year <year1> --end-year <year2>
+# to generate your target jobscript (again --help for instructions):
+> ./gen_mksurfdata_jobscript_single.py --number-of-nodes 24 --tasks-per-node 12 --namelist-file target.namelist
+> qsub mksurfdata_jobscript_single
+
+=======================
+running for the generation of multiple datasets
+=======================
+# Note that gen_mksurfdata_jobscript_multi.py runs ./gen_mksurfdata_namelist.py
+> ./gen_mksurfdata_jobscript_multi.py --help
+> ./gen_mksurfdata_jobscript_multi.py --number-of-nodes 24 --tasks-per-node 12 --scenario all
+> qsub mksurfdata_jobscript_multi
+
+================
+NOTES
+================
+
+------------------------------------------------------------
+slevis HAS RUN THESE CASES and HAS LISTED ISSUES ENCOUNTERED
+------------------------------------------------------------
+1) 1x1_brazil good except:
+ time = -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 0 ;
+ instead of
+ time = 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12 ;
+
+2) 10x15 mostly good BUT:
+- bigger diffs than I would like for NAT_PFT, though .log file shows global PFT
+  areas same => Gridcell-level diffs due to mapping files? Confirm we have no
+  discrepancy between .log and .nc files
+- LAI in .log comes out shifted by one
+- Soil color, ag fire peak month outputs too high in .log
+- Soil txt, peat, fmax, lakes, harvest too low in .log unless raw data changed
+- Slope, stddev, soildepth, urban, glaciers not appearing in .log
+- VOC em factors from input not appearing in .log
+
+3) --merge_gis, --nocrop work
+4) --hires_pft works, but must abort for years /= 2005 (and 1850?)
+5) --potveg_flag needs file(s) converted to cdf5
+6) --fast doesn't do anything as far as I can tell; do we need it?
```

Comment:
TODO Need to remove --fast option

This list in the README includes more TODOs.

Reply:
@slevisconsulting what the fast option did before was to set STD_ELEV and SLOPE to fixed values so that you didn't have to process the 1km topo dataset. This didn't matter much for mksurfdata_map, but it did have a huge impact on the creating of the mapping files, and if you didn't create a 1km mapping file -- then you had to set STD_ELEV and SLOPE to constants. So if it was triggered in the first it had to be triggered the same in mksurfdata_map.

We are adding special handling of the 1km dataset which is different now, so you either need a flag to trigger it (or as @mvertens suggests) it looks for file availability and uses other datasets if they are available.